Sources/MessagingPush/MessagingPush.swift

```diff
@@ -0,0 +1,122 @@
+import CioTracking
+import Foundation
+
+/**
+ Swift code goes into this module that are common to *all* of the Messaging Push modules (APN, FCM, etc).
+ So, performing an HTTP request to the API with a device token goes here.
+  */
+open class MessagingPush {
```

Comment:
I know that you linked to apple docs stating they recommend you do not persist the device token. 

However, since we are storing it in memory, do we know for sure that APN will call `didRegisterForRemoteNotificationsWithDeviceToken` for us the next time that the app is restarted? 

I am nervous that there will be a scenario where a device token is valid for a device but `deviceToken` is still nil in this class.

Reply:
from my reading, as long as the app calls `registerForRemoteNotificationTypes`, the `didRegisterForRemoteNotificationsWithDeviceToken` delegate will get called every time it is opened, also gets called on settings change as well. I vote we stick with the Apple best practice for the time being and then do a few tests. Actually persisting the device token is also pretty trivial, so if we have any concern after testing that the token doesn't show up or something then we can raise it at that point.

https://stackoverflow.com/questions/10983536/when-is-didregisterforremotenotificationswithdevicetoken-called

Reply:
I agree to follow what they mention. I have never tested this myself before (I have always used a background queue in an app that persisted this data for me) so I am not sure on the behavior. 

QA will help assert this is true. 

I am working with QA right now to help them test the behavior of the SDK better. I am thinking we create log statements throughout the SDK (this is a feature of the SDK for customers, not just QA so helping everyone!) and so QA can check the logs of the RH app and see "device token registered" when app starts up. Then we can confirm that this fact is true.